specification/chaos/resource-manager/Microsoft.Chaos/preview/2023-01-31-preview/types/applications.json

```diff
@@ -0,0 +1,360 @@
+      "required": [
+        "clientId",
+        "tenantId",
+        "type"
+      ]
+    },
+    "containingFunctionProperties": {
+      "type": "object",
+      "properties": {
+        "signature": {
+          "$ref": "#/definitions/signatureProperties",
+          "description": "Method signature of the containing function"
+        }
+      },
+      "description": "Defines the signature of the function calling the faultable function",
+      "required": [
+        "signature"
+      ]
+    },
+    "faultableFunctionProperties": {
+      "type": "object",
+      "properties": {
+        "signature": {
+          "$ref": "#/definitions/signatureProperties",
+          "description": "Method Signature of the faultable function"
+        },
+        "category": {
+          "type": "string",
+          "description": "SDK or Custom library in which fault is injected"
+        },
+        "faultCapabilities": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/faultCapabilityProperties"
+          },
+          "description": "faults supported by torch for the function"
+        }
+      },
+      "description": "Application function in which fault can be injected",
+      "required": [
+        "signature"
+      ]
+    },
+    "faultCapabilityProperties": {
+      "type": "object",
+      "properties": {
+        "faultUrn": {
+          "type": "string",
+          "description": "URN for fault capability"
+        },
+        "parameters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/faultParameter"
+          },
+          "description": "Input Parameters supported by a fault"
+        }
+      },
+      "description": "Capabilities supported by the faults"
+    },
+    "faultParameter": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "name of a fault parameter"
+        }
+      }
+    },
+    "sourceProperties": {
+      "type": "object",
+      "properties": {
+        "codePath": {
+          "type": "string",
+          "description": "code path where function is defined"
+        },
+        "startLine": {
+          "type": "integer",
+          "format": "int32",
+          "description": "line in the code where function starts"
+        },
+        "startColumn": {
+          "type": "integer",
+          "format": "int32",
+          "description": "column in the code where function starts"
+        },
+        "endLine": {
+          "type": "integer",
+          "format": "int32",
+          "description": "line in the code where function ends"
+        },
+        "endColumn": {
+          "type": "integer",
+          "format": "int32",
+          "description": "column in the code where function ends"
+        }
+      },
+      "description": "defines the function code path and location in the class"
+    },
```

Comment:
Yikes. Is this going to be able keep itself up to date on line numbers and stuff as the code changes?

Reply:
This api is going to read the metadata from our data plane api which will get the update line numbers reported by our agents. This data is only readonly data for applications api and Create/Update operation will not be supporting any update of this data.

Reply:
Ah OK. Want to mark them as 'readOnly' if they'll never be writable?